File: Steven/src/main/java/edu/parser/nodes/expression/Equal.java
package edu.parser.nodes.expression;

import edu.parser.Visitor;
import edu.parser.nodes.AbstractNode;

/**
 * Created by Steven Kok on 21/02/2015.
 */
public class Equal extends BinaryExpression {
    public Equal(Expression left, Expression right) {
        super(left, right);
    }

    @Override
    public AbstractNode accept(Visitor visitor) {
        return visitor.visit(this);
    }

    @Override
    public boolean hasBooleanOperands() {
        return false;
    }
}

File: fabien/src/QL/operators/Equal.py

from BooleanOperand import BooleanOperand

class Equal(BooleanOperand):
    def evaluate(self, answers):
        return self.left.evaluate(answers) == self.right.evaluate(answers)

    def __repr__(self):
        return "%s == %s" % (self.left, self.right)
